addons/sourcemod/scripting/influx_simpleranks_chat.sp

```diff
@@ -36,107 +70,91 @@ public APLRes AskPluginLoad2( Handle hPlugin, bool late, char[] szError, int err
     RegPluginLibrary( INFLUX_LIB_SIMPLERANKS_CHAT );
 }
 
-public void OnPluginStart()
+public Action CP_OnChatMessage(int& author, ArrayList recipients, char[] flagstring, char[] name, char[] message, bool & processcolors, bool & removecolors)
 {
-    // CONVARS
-    g_ConVar_PrintToServer = CreateConVar( "influx_simpleranks_chat_printtoserver", "2", "Is the player's message printed to server console? 1 = Do it, 2 = Remove player rank as well.", FCVAR_NOTIFY );
-    
-    AutoExecConfig( true, "simpleranks_chat", "influx" );
-    
-    
-    // LIBRARIES
-    g_bLib_BaseComm = LibraryExists( "basecomm" );
-    g_bLib_SilentChatCmds = LibraryExists( INFLUX_LIB_SILENT_CHATCMDS );
-    
-    
-    
-    g_bHasBaseChat = ( FindPluginByFile( "basechat.smx" ) != null ) ? true : false;
+    if( !IsFakeClient( author ) )
+    {
+        char szRank[64];
+
+        GetClientRank( author, szRank, sizeof( szRank ), false );
+
+        if( szRank[0] )
+        {   
+            Format(name, MAXNAME_LENGTH, "%s %s", szRank, name );
+
+            return Plugin_Changed;
+        }
+        
+    }
+
+    return Plugin_Continue;
 }
 
-public void OnLibraryAdded( const char[] lib )
+public Action OnChatMessage(int &author, Handle recipients, char[] name, char[] message)
 {
-    if ( StrEqual( lib, "basecomm" ) ) g_bLib_BaseComm = true;
-    if ( StrEqual( lib, INFLUX_LIB_SILENT_CHATCMDS ) ) g_bLib_SilentChatCmds = true;
+    if( !IsFakeClient( author ) )
+    {
+        char szRank[64];
+
+        GetClientRank( author, szRank, sizeof( szRank ), false );
+
+        if( szRank[0] )
+        {   
+            Format(name, MAXNAME_LENGTH, "%s %s", szRank, name );
+
+            return Plugin_Changed;
+        }
+        
+    }
+
+    return Plugin_Continue;
 }
 
-public void OnLibraryRemoved( const char[] lib )
+int MType;
+
+public void cc_proc_MsgBroadType(const int iType)
 {
-    if ( StrEqual( lib, "basecomm" ) ) g_bLib_BaseComm = false;
-    if ( StrEqual( lib, INFLUX_LIB_SILENT_CHATCMDS ) ) g_bLib_SilentChatCmds = false;
+    MType = iType;
 }
 
-public Action OnClientSayCommand( int client, const char[] szCommand, const char[] szMsg )
```

Comment:
influx_simpleranks_chat won't work without a chat processor.

Add OnClientSayCommand and all it's funtionality back and return early if a chat processor library exists.